src/PhpSpec/Process/Shutdown/Shutdown.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Process\Shutdown;
+
+class Shutdown
+{
+    protected $actions;
+
+    public function __construct()
+    {
+        $this->actions = array();
+    }
+
+    public function registerShutdown()
+    {
+        ini_set('display_errors', '0');
+        error_reporting(E_ERROR);
+        register_shutdown_function(array($this, 'runShutdown'));
+    }
+
+    public function registerAction(ShutdownActionInterface $action)
+    {
+        $this->actions[] = $action;
+    }
+
+    public function count()
+    {
+        return count($this->actions);
+    }
+
+    public function runShutdown()
+    {
+        foreach ($this->actions as $shutdownActions) {
```

Comment:
I've been thinking about these ShutdownActions...

Maybe `shutdownAction` is too generic (although we may need it in future); perhaps they should be `fatalErrorAction`s?

That way you could move the `get_last_error` logic into this method, and explicitly pass the error details through to the actions when they're run, otherwise I can see us having repetition once we get more than one fatal handler.